### PR TITLE
Fix #1785 ConstField empty javadoc

### LINKED ISSUE #1785: XJC ConstField generates empty javadoc comment blocks
As of v4.0.4, the [ConstField](https://github.com/eclipse-ee4j/jaxb-ri/blob/4.0.4-RI/jaxb-ri/xjc/src/main/java/com/sun/tools/xjc/generator/bean/field/ConstField.java) class generates empty Javadoc comment blocks when no `<javadoc>text</javadoc>` is explicitly provided in the XML schema.

**Example: Empty Javadoc for a Constant Field**
~~~
    /**
     * 
     * 
     */
    @XmlAttribute(name = "fa_long")
    public static final boolean FA_LONG = 7L;
~~~

A fix is to conditionally append the `prop.javadoc` only when it has content:

~~~
--- a/jaxb-ri/xjc/src/main/java/com/sun/tools/xjc/generator/bean/field/ConstField.java
+++ b/jaxb-ri/xjc/src/main/java/com/sun/tools/xjc/generator/bean/field/ConstField.java
@@ -54,8 +54,11 @@ final class ConstField extends AbstractField {
 
         $ref = outline.ref.field(JMod.PUBLIC|JMod.STATIC|JMod.FINAL,
             ptype!=null?ptype:implType, prop.getName(true), defaultValue );
-        $ref.javadoc().append(prop.javadoc);
         
+        // Do not append empty javadoc.
+        if ( (prop.javadoc != null) && (prop.javadoc.length() > 0) )
+            $ref.javadoc().append(prop.javadoc);
+
         annotate($ref);
     }
~~~

**Rationale**

Currently, when `$ref.javadoc()` is invoked by `ConstField`, the `jdoc` field of `JFieldVar` is initialized to a non-null value. Later, the `declare` method is invoked and the `JFormatter.g(jdoc)` method is triggered whenever `jdoc` is not null. This leads to a call to [`JDocComment.generate(JFormatter)`](https://github.com/eclipse-ee4j/jaxb-ri/blob/75b0b973d410f512914b38ed079a5b0641b6d480/jaxb-ri/codemodel/codemodel/src/main/java/com/sun/codemodel/JDocComment.java#L142) which includes blank lines when `@param`, `@return`, `@throws`, etc. are not applicable.

The proposed fix is to prevent an unnecessary initialization of the `jdoc` field of `JFieldVar`; thus, avoiding a call to `JDocComment.generate(JFormatter)`. I will submit a [pull request](https://github.com/eclipse-ee4j/jaxb-ri/pull/1786) with the fix and a unit test.

**ConstField: Call Sequence**
~~~
com.sun.codemodel.JFieldVar.javadoc()
	...
    public JDocComment javadoc()
	{
        if( jdoc == null ) 
            jdoc = new JDocComment(owner.owner());
        return jdoc;
    }
	...
com.sun.codemodel.JDefinedClass.declareBody(JFormatter)
com.sun.codemodel.JFormatter.d(JDeclaration)
com.sun.codemodel.JFieldVar.declare(JFormatter)
	...
    public void declare(JFormatter f)
	{
        if( jdoc != null )
            f.g( jdoc );
        super.declare( f );
    }
	...
com.sun.codemodel.JFormatter.g(JGenerable)
com.sun.codemodel.JDocComment.generate(JFormatter)
~~~

### PR DESCRIPTION
**Pull request by patrodyne**
This pull request resolves Issue #1785.

### CODE REVIEW

**Review by lukasj — COMMENTED** [review]
Can you update the copyright year in changed files to say 2024 (instead of 2022, 2023), please? I also prefer if static imports are avoided.

Thank you.

**Review by lukasj — COMMENTED** [review]
it would be also good to add a note to the [changelog](https://github.com/eclipse-ee4j/jaxb-ri/blob/master/jaxb-ri/docs/release-documentation/src/docbook/jaxb-changelog.xml#L31)

**Comment by patrodyne** [comment]
I've pushed commits:

+ [Updated copyright to 2024](https://github.com/eclipse-ee4j/jaxb-ri/pull/1786/commits/6719ac9a3f7c621529a09c2201a03f6fc0a27de4)
+ [Refactored static imports](https://github.com/eclipse-ee4j/jaxb-ri/pull/1786/commits/ebc6e7fd4ac13f46afcdcc878cea1c11e7c50e21)
+ [Added #1785 to changelog](https://github.com/eclipse-ee4j/jaxb-ri/pull/1786/commits/33548d433ed31c4f8afedf71810e155101e61e66)

File: jaxb-ri/xjc/src/main/java/com/sun/tools/xjc/generator/bean/field/ConstField.java [hunk 1]
```diff
@@ -1,5 +1,5 @@
 /*
- * Copyright (c) 1997, 2022 Oracle and/or its affiliates. All rights reserved.
+ * Copyright (c) 2024 Oracle and/or its affiliates. All rights reserved.
```
Comment (lukasj) [inline jaxb-ri/xjc/src/main/java/com/sun/tools/xjc/generator/bean/field/ConstField.java:2]: should be _1997, 2024_

File: jaxb-ri/xjc/src/test/java/com/sun/tools/xjc/CodeGenTest.java [hunk 2]
```diff
@@ -1,5 +1,5 @@
 /*
- * Copyright (c) 2021, 2023 Oracle and/or its affiliates. All rights reserved.
+ * Copyright (c) 2024 Oracle and/or its affiliates. All rights reserved.
```
Comment (lukasj) [inline jaxb-ri/xjc/src/test/java/com/sun/tools/xjc/CodeGenTest.java:2]: should be _2021, 2024_

**Review by lukasj — APPROVED** [review]
Thank you!

**Pull request merged**